### PR TITLE
Add KraftKit configuration file for KraftCloud

### PR DESCRIPTION
**Pull request by razvand**
Add `kraft.cloud.yaml` configuration file to build and run for KraftCloud.

Build with:

```console
kraft build --kraftfile kraft.cloud.yaml -j $(nproc)
```

Run with:

```console
sudo kraft net create -n 172.44.0.1/24 kraft0
sudo kraft run --kraftfile kraft.cloud.yaml --network bridge:kraft0 --plat firecracker --initrd ./fs0 -a netdev.ipv4_addr=172.44.0.2 -a netdev.ipv4_gw_addr=172.44.0.1 -a netdev.ipv4_subnet_mask=255.255.255.0 -- -c /nginx/conf/nginx.conf
```

### CODE REVIEW

File: kraft.cloud.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    - CONFIG_LIBUKSCHEDCOOP=y
+    - CONFIG_LIBPOSIX_LIBDL=y
+    - CONFIG_LIBPOSIX_PROCESS=y
+    - CONFIG_LIBPOSIX_USER=y
+    - CONFIG_UKSYSINFO=y
+    - CONFIG_LIBUKSIGNAL=y
+    - CONFIG_LIBUKMMAP=y
+    - CONFIG_LIBUKTIME=y
+targets:
+  - name: firecracker-x86_64
```
Comment (marcrittinghaus) [inline kraft.cloud.yaml]: Rename to `kraftcloud-x86_64`

File: kraft.cloud.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    kconfig:
+      - CONFIG_LWIP_UKNETDEV=y
+      - CONFIG_LWIP_TCP=y
+      - CONFIG_LWIP_WND_SCALE=y
+      - CONFIG_LWIP_TCP_KEEPALIVE=y
+      - CONFIG_LWIP_THREADS=y
+      - CONFIG_LWIP_SOCKET=y
+      - CONFIG_LWIP_AUTOIFACE=y
+      - CONFIG_LWIP_IPV4=y
+      - CONFIG_LWIP_DHCP=y
```
Comment (marcrittinghaus) [inline kraft.cloud.yaml]: We don't need (even want!) DHCP

**Review by marcrittinghaus — CHANGES_REQUESTED** [review]
Hey @razvand,

please check out the requested changes and also fix the commit title.

We probably also need to switch to `posix-mmap` due to `lib-ukp-bin`

**Comment by craciunoiuc** [comment]
only one comment from my side: You can use maps instead of lists if you fancy those

`- CONFIG_LWIP_DHCP=y` -> `  CONFIG_LWIP_DHCP: "y"`

**Comment by StefanJum** [comment]
@razvand running the commands you mentioned, I get this error at runtime:
`could not detect any project targets based on plat="fc" arch="x86_64"`.

Everything build fine, the application image is there. Any idea?

**Comment by razvand** [comment]
> @razvand running the commands you mentioned, I get this error at runtime: `could not detect any project targets based on plat="fc" arch="x86_64"`.
> 
> Everything build fine, the application image is there. Any idea?

You need the latest [`kraftkit`](https://github.com/unikraft/kraftkit). Download it, do `make devenv`, then `make kraft`, the result executable is part of `dist/kraft`.

**Review by StefanJum — APPROVED** [review]
All good, thanks.

Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Review by craciunoiuc — APPROVED** [review]
All good here. Thanks!

Reviewed-by: Cezar Craciunoiu <cezar.craciunoiu@unikraft.io>

**Review by marcrittinghaus — APPROVED** [review]
Approved-by: Marc Rittinghaus <marc.rittinghaus@unikraft.io>

**Pull request closed without merging**